packages/strapi-utils/lib/buildQuery.js

```diff
@@ -108,9 +108,11 @@ const buildQuery = ({ model, filters = {}, ...rest }) => {
         const { type } = _.get(assocModel, ['attributes', attribute], {});
 
         // cast value or array of values
-        const castedValue = Array.isArray(value)
-          ? value.map(val => castValueToType({ type, value: val }))
-          : castValueToType({ type, value: value });
+        const castedValue = operator == 'null'
+          ? castValueToType({ type: 'boolean', value: value })
```

Comment:
can you plz just call `castValueToType({ type, value, operator })` and do the  test in the function directly :)